### PR TITLE
[Onboarding Flow] Added Page For Archived User

### PR DESCRIPTION
**Pull request by vinit717**
##### 

- [ ] #575 

### What is the change?

Created archived card for archived user

### Is Development Tested?

- [ ] Yes
- [ ] No

### video
https://github.com/Real-Dev-Squad/website-www/assets/111434418/a275d6d9-bd0b-4f97-85b5-6461333398c1

In the middle of the video, I have changed archived to false

### CODE REVIEW

File: app/components/stepper.hbs [hunk 1]
```diff
@@ -27,66 +27,77 @@
...
-  {{else if (eq this.currentStep 2)}}
-    <JoinSteps::StepTwo @setIsPreValid={{this.setIsPreValid}} @isValid={{this.isValid}} @setIsValid={{this.setIsValid}}/>
-  {{else if (eq this.currentStep 3)}}
-    <JoinSteps::StepThree @setIsPreValid={{this.setIsPreValid}} @isValid={{this.isValid}} @setIsValid={{this.setIsValid}}/>
-  {{else if (eq this.currentStep 4)}}
-    <JoinSteps::StepFour login={{this.login}} />
-  {{else if (eq this.currentStep 5)}}
-    <JoinSteps::StepFive />
-  {{/if}}
+  {{#if (and (eq this.currentStep 0) (not this.login.isLoggedIn))}}
```
Comment (shubhamsigdar1) [inline app/components/stepper.hbs:30]: Don intrupt the current stepper.hbs file
I created stepper-signup.hbs component- 
https://github.com/Real-Dev-Squad/website-www/pull/607/files#diff-7bad8c77216aba9edbc363d2c213847ebf39040a04b5c9ade78419fca26b8584R1

@rohan09-raj suggested above idea

File: app/components/stepper.js [hunk 2]
```diff
@@ -64,6 +67,19 @@ export default class StepperComponent extends Component {
     this.isValid = false;
   }
 
+  @action logingithub() {
```
Comment (shubhamsigdar1) [inline app/components/stepper.js:70]: same as above comment: I created  stepper-signup.js
https://github.com/Real-Dev-Squad/website-www/pull/607/files#diff-7b44f6d0028b6c487af61294485dd802b40c1b32d0b182caf59638cc2241d7b1R1

File: app/templates/join.hbs [hunk 3]
```diff
@@ -1,2 +1,14 @@
...
-<Stepper />
\ No newline at end of file
+{{#if this.login.isLoading}}
+<div data-test-loading class='intro__loading'>
+    <Fa-Icon @size='2x' @icon='circle-notch' @spin={{true}} />
+</div>
+{{else}}
+{{#if this.login.userData.roles.archived}}
+<Card>
+<h3 class="archived__heading">Archived User</h3>
```
Comment (shubhamsigdar1) [inline app/templates/join.hbs:9]: Why not create seperate component

File: app/templates/join.hbs [hunk 4]
```diff
@@ -1,2 +1,14 @@
 {{page-title 'Join'}}
-<Stepper />
\ No newline at end of file
+{{#if this.login.isLoading}}
+<div data-test-loading class='intro__loading'>
```
Comment (shubhamsigdar1) [inline app/templates/join.hbs:3]: login button page in component signup-steps/step-zero
like this step-one page for firstname,lastname,username- 
https://github.com/Real-Dev-Squad/website-www/pull/607/files#diff-a04f9bf1d30ebf7d780a7ccd09c47c973d988ed678833dea68f1e1b31c4352c9R1

**Review by shubhamsigdar1 — COMMENTED** [review]
Hello @vinit717 where you handle this case: 
when user is archieved and when we go to /join route then it will redirect to /archived route
and when user is not archieved and when we go to /archived then it redirect to /join route

**Comment by vinit717** [comment]
> Hello @vinit717 where you handle this case: when user is archieved and when we go to /join route then it will redirect to /archived route and when user is not archieved and when we go to /archived then it redirect to /join route

Because currently that approach is not take cape of every edge case

File: app/templates/join.hbs (hunk 3 above)
Comment (vinit717) [reply app/templates/join.hbs:9]: I have created a separate component for the card but for archived it is a very small description so it depends to create separate or not

File: app/routes/join.js [hunk 5]
```diff
@@ -1,3 +1,6 @@
 import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
 
-export default class JoinRoute extends Route {}
+export default class JoinRoute extends Route {
+  @service login;
```
Comment (shubhamsigdar1) [inline app/routes/join.js:5]: Why you calling service login here??

File: app/templates/join.hbs (hunk 4 above)
Comment (shubhamsigdar1) [reply app/templates/join.hbs:3]: Talked with @rohan09-raj said its fine

File: tests/integration/components/stepper-test.js [hunk 6]
```diff
@@ -6,6 +6,16 @@ import { hbs } from 'ember-cli-htmlbars';
...
 
+  hooks.beforeEach(function () {
+    this.owner.register(
+      'service:login',
+      class MockLoginService {
+        isLoggedIn = false;
+      }
+    );
+    this.login = this.owner.lookup('service:login');
+  });
```
Comment (shubhamsigdar1) [inline tests/integration/components/stepper-test.js:17]: Can you explain what you trying to do here

**Review by shubhamsigdar1 — COMMENTED** [review]
Great work @vinit717 keep it up

File: app/routes/join.js (hunk 5 above)
Comment (vinit717) [reply app/routes/join.js:5]: Because to access the role user must be login

File: tests/integration/components/stepper-test.js (hunk 6 above)
Comment (vinit717) [reply tests/integration/components/stepper-test.js]: Removed

File: app/components/card.hbs [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<div class="card__reusable">
```
Comment (satyam73) [inline app/components/card.hbs:1]: we generally add __ for childs under the block in `BEM` convention

Comment (satyam73) [inline app/components/card.hbs:1]: we generally add `__` for childs under the block in BEM convention.

It's like,
`block__element--modifier`

example:
`card__childElementOfCard--md`

File: app/styles/card.module.css [hunk 8]
```diff
@@ -0,0 +1,19 @@
+.card__reusable{
+    min-height: 60vh;
+    max-width: 40vw;
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    border: 2px solid var(--color-lightgrey);
+    margin: 80px auto;
```
Comment (satyam73) [inline app/styles/card.module.css:9]: That much margin, can we an alternative here?

File: app/styles/card.module.css [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    background-color: var(--color-cultured);
+    border-radius: 10px;
+}
```
Comment (satyam73) [inline app/styles/card.module.css:12]: try adding a line spacing here

File: app/styles/card.module.css [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+@media screen and (max-width: 625px) {
```
Comment (satyam73) [inline app/styles/card.module.css:13]: try adding a line spacing here

File: app/styles/intro.module.css [hunk 11]
```diff
@@ -43,6 +43,10 @@
   text-align: justify;
 }
 
+.archived__heading{
+font-size: 24px;
```
Comment (satyam73) [inline app/styles/intro.module.css:47]: Let's use rem as a unit for fonts

File: tests/integration/components/card-test.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+import { setupRenderingTest } from 'website-www/tests/helpers';
+import { render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | card', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(hbs`<Card />`);
+    assert.dom('.card__reusable').exists();
```
Comment (satyam73) [inline tests/integration/components/card-test.js:11]: use `data-test` selectors for picking up the dom elements

File: tests/integration/components/card-test.js
Comment (satyam73) [inline tests/integration/components/card-test.js:1]: use data-test selectors for picking up the dom elements

File: app/components/card.hbs
Comment (satyam73) [inline app/components/card.hbs:1]: there is also one card component exists, make sure your naming of component and css classnames doesn't collide with that one

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: app/templates/join.hbs (hunk 4 above)
Comment (rohan09-raj) [inline app/templates/join.hbs:3]: Can we please center the loader here and have some min height for this page ?
Check /intro/:userId route here
image

Comment (vinit717) [reply app/templates/join.hbs:4]: done

**Review by rohan09-raj — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]

**Pull request merged**